tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Azure/Functions/AzureFunctionsHostDetector.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="AzureFunctionsHostDetector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using Datadog.Trace.Logging;
+
+internal static class AzureFunctionsHostDetector
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AzureFunctionsHostDetector));
+
+    public static readonly bool IsRunningInFunctionsHost;
+
+    static AzureFunctionsHostDetector()
+    {
+        IsRunningInFunctionsHost = DetectIfFunctionsHost();
+
+        if (IsRunningInFunctionsHost)
+        {
+            Log.Information("Detected Azure Functions host process.");
+        }
+    }
+
+    private static bool DetectIfFunctionsHost()
+    {
+        try
+        {
+            var functionsRuntime = Environment.GetEnvironmentVariable("FUNCTIONS_WORKER_RUNTIME");
+            if (string.IsNullOrEmpty(functionsRuntime))
+            {
+                return false; // we are the worker process, not the host
```

Comment:
I think `FUNCTIONS_WORKER_RUNTIME` is set on both the host and worker processes?

Reply:
Btw we have some of these checks here:
https://github.com/DataDog/dd-trace-dotnet/blob/835d79f94261ca1cbf5177c50ebc442329bbb057/tracer/src/Datadog.Trace/Util/EnvironmentHelpers.cs#L110-L121

Reply:
Ah yes, my comment here is not correct and should be that we aren't in AzureFunctions, but thanks for the EnvironmentHelpers link, I'm shifting stuff to there as it makes more sense